examples/demo/package.json

```diff
@@ -19,7 +19,7 @@
         "react-admin": "^2.0.0",
         "react-app-polyfill": "^0.1.3",
         "react-dom": "~16.8.0",
-        "react-redux": "^7.1.0-rc.1",
+        "react-redux": "^7.1.0",
```

Comment:
Shouldn't we remove this line as `ra-core` already includes it as a dependency (not a peer)?

Reply:
No, because some components in ra-ui-material-ui use useDispatch and useSelector directly.

Reply:
And ? `ra-ui-material` also depends on `ra-core`

Reply:
Right, but we have to be explicit about our dependencies. If ra-ui-material-ui imports functions from recat-router, it must have react-router in dependencies. I didn't invent that rule, it's enforced by TypeScript I think. 

Also, this PR doesn't change anything regarding new dependencies - it just upgrades them.